rope/base/prefs.py

```diff
@@ -1,30 +1,220 @@
+"""Rope preferences."""
+from dataclasses import asdict, dataclass, fields
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+from packaging.requirements import Requirement
+from pytoolconfig import PyToolConfig, UniversalKey, field
+from pytoolconfig.sources import Source
+
+from rope.base.resources import Folder
+from rope.base.utils import pycompat
+
+
+@dataclass
 class Prefs:
-    def __init__(self):
-        self.prefs = {}
-        self.callbacks = {}
+    """Class to store rope preferences."""
+
+    ignored_resources: List[str] = field(
+        default_factory=[
+            "*.pyc",
+            "*~",
+            ".ropeproject",
+            ".hg",
+            ".svn",
+            "_svn",
+            ".git",
+            ".tox",
+            ".venv",
+            "venv",
+        ].copy,
```

Comment:
This is a bit too clever to use `.copy()` here like this, should use `lambda` instead.

Reply:
Done